File: com/problem/java/googleKickstart/2022/session/3/RecordBreakerStarter.java
import java.util.Scanner;

class Solution 
{
  public static int numberOfRecordBreakingDays(int numberOfDays, int[] visitors)
  {
    int recordBreakingDays = 0;
    int previousRecord = 0;

    for (int j = 0; j < visitors.length; ++j)
    {
      Boolean greaterThanPreviousDays = j == 0 || visitors[j] > previousRecord;
      Boolean greaterThanNextDays = j == visitors.length-1 || visitors[j] > visitors[j+1];
      
      if (greaterThanPreviousDays && greaterThanNextDays)
      {
        recordBreakingDays++;
      }
        previousRecord = Math.max(previousRecord, visitors[j]);
    }
    
    return recordBreakingDays;
  }

  public static void main(String[] args)
  {
    Scanner scanner = new Scanner(System.in);
    // Get number of test cases in input
    int testCaseCount = scanner.nextInt();
    // Iterate through test cases
    for (int tc = 1; tc <= testCaseCount; ++tc)
    {
      // Get number of days for this test case
      int numberOfDays = scanner.nextInt();
      // Get attendance for each day
      int[] visitorsPerDay = new int[numberOfDays];
      for(int d = 0; d < numberOfDays; ++d)
      {
        visitorsPerDay[d] = scanner.nextInt();
      }
      // Print results
      int answer = numberOfRecordBreakingDays(numberOfDays, visitorsPerDay);
      System.out.println("Case #"+tc+": "+answer);
    }
  }
}

File: com/problem/cpp/googleKickStart/session/3/RecordBreakerStarter.cpp
#include <bits/stdc++.h>

using namespace std;

int NumberOFRecordBreakingDays(int numberOfDays, vector<int> visitors)
{
  int recordBreakersCount = 0;
  int previousRecord = 0;

  for (int j = 0; j < visitors.size(); ++j)
  {
    bool greaterThanPreviousDays = j == 0 || visitors[j] > previousRecord;
    bool greaterThanNextDays = j == visitors.size() - 1 || visitors[j] > visitors[j + 1];

    if (greaterThanPreviousDays && greaterThanNextDays)
    {
      recordBreakersCount++;
      previousRecord = max(previousRecord, visitors[j]);
    }
  }
  return recordBreakersCount;
}

int main()
{
  int T;
  cin >> T;
  for (int tc = 1; tc <= T; ++tc)
  {
    int N;
    cin >> N;
    vector<int> V(N);
    for (int i = 0; i < N; ++i)
    {
      cin >> V[i];
    }
    cout << "Case #" << tc << ": " << NumberOFRecordBreakingDays(N,V) << endl;
  }
  return 0;
}
